tla/consensus/ccfraft.tla

```diff
@@ -1260,6 +1394,24 @@ LogConfigurationConsistentInv ==
                     log[i][idx].contentType = TypeReconfiguration 
                     => configurations[i][idx] = log[i][idx].configuration
 
+\* Check each node's retirement phase is consistent with its local state
+RetirementPhaseConsistentInv ==
+    \A i \in Servers :
+        \/ /\ retirementPhase[i] = NotRetiring 
+           /\ RetirementIndex(i) = 0
+        \* RetirementOrdered - node's retirement in its log
+        \/ /\ retirementPhase[i] = RetirementOrdered 
+           /\ RetirementIndex(i) # 0
+        \* RetirementSigned - node' retirement is signed 
+        \/ /\ retirementPhase[i] = RetirementSigned
+           /\ RetirementIndex(i) # 0
+           /\ RetirementIndex(i) <= MaxCommittableIndex(log[i])
```

Comment:
We could check that if the node is leader and its current term matches the term of retirement index, there are no other signatures between RetirementIndex(i) and MaxCommittableIndex(log[i]). But that's a fairly minor detail.

Reply:
I don't think that is true in the spec at the moment, should it be?

Reply:
I think it may be, because of https://github.com/microsoft/CCF/blob/37ec70f004fba7c10c4dd65c4d69d81f097c3b75/src/consensus/aft/raft.h#L576C13-L576C39, and the logic that tries to protect against consecutive signatures in the same term. But you're right that we should probably have a closer look first.

Reply:
My question was why would the sig be consecutive? I think in the impl the leader continues to accept new requests until its retirement is committed

Reply:
The line above is in replicate(), rejecting new entries as soon as retirement_committable_idx is set (i.e. as soon as a signature was emitted after that leader's retirement tx, modulo rollbacks, but then the term would be higher).